src/Http/Routing/src/Builder/DelegateEndpointConventionBuilder.cs

```diff
@@ -13,12 +13,22 @@ public sealed class DelegateEndpointConventionBuilder : IEndpointConventionBuild
     {
         private readonly List<IEndpointConventionBuilder> _endpointConventionBuilders;
 
-        internal DelegateEndpointConventionBuilder(IEndpointConventionBuilder endpointConventionBuilder)
+        /// <summary>
+        /// Instantiates a new <see cref="DelegateEndpointConventionBuilder" /> given a single
+        /// <see cref="IEndpointConventionBuilder" />.
+        /// </summary>
+        /// <param name="endpointConventionBuilder">The <see cref="IEndpointConventionBuilder" /> to instantiate with.</param>
+        public DelegateEndpointConventionBuilder(IEndpointConventionBuilder endpointConventionBuilder)
         {
             _endpointConventionBuilders = new List<IEndpointConventionBuilder>() { endpointConventionBuilder };
         }
 
-        internal DelegateEndpointConventionBuilder(List<IEndpointConventionBuilder> endpointConventionBuilders)
+        /// <summary>
+        /// Instantiates a new <see cref="DelegateEndpointConventionBuilder" /> given multiple
+        /// <see cref="IEndpointConventionBuilder" /> instances.
+        /// </summary>
+        /// <param name="endpointConventionBuilders">A list of <see cref="IEndpointConventionBuilder" /> instances.</param>
+        public DelegateEndpointConventionBuilder(List<IEndpointConventionBuilder> endpointConventionBuilders)
```

Comment:
How about `IEnumerable` instead of `List` if the list that is passed here is not supposed to be changed later?
```suggestion
        public DelegateEndpointConventionBuilder(IEnumerable<IEndpointConventionBuilder> endpointConventionBuilders)
```